R/age_group.R

```diff
@@ -0,0 +1,60 @@
+#' Create age groups
+#'
+#' @description
+#' A wrapper for \code{\link[base]{cut}} tailored for creating age groups. For
+#' each value of integer \code{x}, what age group does it fall into.
+#'
+#' @param x a vector of numeric values
+#' @param from,to,by start, end and increment values for a sequence of age
+#'   groups.
+#' @param as_factor The default behaviour is to return a character vector. Use
+#'   \code{TRUE} to return a factor vector instead.
+#'
+#' @return A character vector, where each element is the age group for the
+#'   corresponding element in \code{x}. If \code{as_factor = FALSE} a factor
+#'   vector is returned instead.
+#'
+#' @details
+#' The \code{from},\code{to} and \code{by} are used to create distinct age
+#' groups \code{from-from+by}, ..., up to \code{to+}, or a value less than
+#' \code{to} if \code{to} is not an exact multiple. The dafault values provide
+#' the standard five year age groups.
```

Comment:
This section is a little difficult to follow. I think I would re-word it to something like:

"The `from`, `to` and `by` values are used to create distinct age groups. `from` dictates the starting age of the lowest age group, and `by` indicates how wide each group should be. `to` stipulates the cut-off point at which all ages equal to or greater than this value should be categorised together in a `to+` group. If the specified value of `to` is not a multiple of `by` then the value of `to` is rounded down to the nearest multiple of `by`. The default values of `from`, `to` and `by` are those which split ages into the [European Standard Population](https://www.opendata.nhs.scot/dataset/standard-populations/resource/edee9731-daf7-4e0d-b525-e4c1469b8f69) age groups."